### PR TITLE
Update the version of the build tools and gradle

### PR DESCRIPTION
**Pull request by yoelglus**
Also removed some unused code and added .gradle to the .gitignore

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -34,8 +34,8 @@ repositories {
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    compile 'com.facebook.react:react-native:0.49.3'
```
Comment (manhluong) [inline android/build.gradle]: I would favor in general to have more relaxed requirements, unless is strictly necessary.

I mean, does this package have any problems if we use it with RN 0.48? Do we already know we will have problems with RN 0.50?

If not, I would put the +.

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Mon Dec 28 10:00:20 PST 2015
+#Fri Oct 13 11:56:42 BST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip
```
Comment (manhluong) [inline android/gradle/wrapper/gradle-wrapper.properties]: Is this upgrade strictly necessary?

Comment (yoelglus) [reply android/gradle/wrapper/gradle-wrapper.properties]: not really - just aligning to the app

File: android/build.gradle (hunk 1 above)
Comment (yoelglus) [reply android/build.gradle]: In 47 they broke the API causing a compilation error. that's why I did this. We can put the wildcard in the 3rd number if you prefer.

Comment (yoelglus) [reply android/build.gradle]: The problem is, if 50 will break this bridge then we will know as soon as we update the RN version and we'll have a dependency on updating this repo before the main repo is updated

Comment (manhluong) [reply android/build.gradle]: Probably I am missing something but if we use 50, then we have to update to compile 'com.facebook.react:react-native:0.50.something' anyway, right?

Comment (yoelglus) [reply android/build.gradle]: Not necessarily. It is up to us which version we use in each one AFAIK... The thing is, even if we upgrade to 50 we might not do that immediately after it was created. In that case we will use 50 here as soon as it is released...

File: android/gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (manhluong) [reply android/gradle/wrapper/gradle-wrapper.properties]: Not a deal breaker but again I would prefer to not restrict libs dependencies if not necessary.

Up to you.

File: android/build.gradle (hunk 1 above)
Comment (manhluong) [reply android/build.gradle]: Ok, not a deal break for me here.

Up to you.

**Comment by manhluong** [comment]
LGTM for me.

**Pull request merged**